src/kernel/src/ucred/mod.rs

```diff
@@ -105,13 +137,49 @@ impl Ucred {
     }
 
     /// See `prison_priv_check` on the PS4 for a reference.
-    fn prison_priv_check(&self) -> Result<(), PrivilegeError> {
-        // TODO: Implement this.
+    fn prison_priv_check(&self, p: Privilege) -> Result<(), PrivilegeError> {
+        if !self.is_jailed() {
+            return Ok(());
+        }
+
+        match p {
+            Privilege::PROC_SETLOGIN
+            | Privilege::VFS_READ
+            | Privilege::VFS_WRITE
+            | Privilege::VFS_ADMIN
+            | Privilege::VFS_EXEC
+            | Privilege::VFS_LOOKUP => Ok(()),
+            _ => todo!("prison_priv_check({p})"),
+        }
+    }
+
+    /// See `prison_check_af` on the PS4 for a reference.
+    pub fn prison_check_address_family(
+        &self,
+        family: AddressFamily,
+    ) -> Result<(), PrisonCheckAfError> {
+        let pr = &self.prison;
+
+        match family {
+            AddressFamily::UNIX | AddressFamily::ROUTE => {}
+            AddressFamily::INET => todo!(),
+            AddressFamily::INET6 => {
+                if pr.flags().intersects(PrisonFlags::IP6) {
+                    todo!()
+                }
+            }
+            _ => {
+                if !pr.allow().intersects(PrisonAllow::ALLOW_SOCKET_AF) {
+                    return Err(PrisonCheckAfError::SocketAddressFamilyNotAllowed(family));
+                }
+            }
+        }
+
```

Comment:
This one should not be here. It is specific to socket so better to move it there.